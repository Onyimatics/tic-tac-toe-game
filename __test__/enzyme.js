import Enzyme, {
    configure, shallow, mount, render,
  } from 'enzyme';
  import Adapter from 'enzyme-adapter-react-16';
  import toJson from 'enzyme-to-json';
  
  configure({ adapter: new Adapter() });
  export { shallow, mount, render, toJson };
  export default Enzyme;