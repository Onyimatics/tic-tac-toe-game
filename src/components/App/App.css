.App {
  background: radial-gradient(circle, #000, #000);
  background: -webkit-radial-gradient(circle, #f44, #000);
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  text-align: center;
}

