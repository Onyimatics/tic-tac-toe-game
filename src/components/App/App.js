import React, { Component } from 'react';
import './App.css';
import Board from '../Board/Board'

class App extends Component {
  render() {
    return (
        <div className="App">
          <h2>Tic Tac Toe</h2>
          <Board></Board>
        </div>
    );
  }
}

export default App;