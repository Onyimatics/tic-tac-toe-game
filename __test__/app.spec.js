import React from 'react';
import toJson from 'enzyme-to-json';
import App from '../src/components/App/App';
import { mount, shallow} from './enzyme';


describe('SAMPLE TEST', () => {
  it('should return a successful sample test ', () => {
    expect(true).toBeTruthy();
  });
  
  it('should render the App component correctly', () => {
    const Wrapper = mount(<App />);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  it('should render header without crashing', () => {
    const Wrapper = shallow(<App />);
    expect(toJson(Wrapper)).toMatchSnapshot();
    expect(Wrapper.find('div')).toHaveLength(1);
    expect(Wrapper.find('h2')).toHaveLength(1);
  })
});