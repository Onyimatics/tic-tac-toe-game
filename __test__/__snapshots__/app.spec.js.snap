// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAMPLE TEST should render header without crashing 1`] = `
<div
  className="App"
>
  <h2>
    Tic Tac Toe
  </h2>
  <Board />
</div>
`;

exports[`SAMPLE TEST should render the App component correctly 1`] = `
<App>
  <div
    className="App"
  >
    <h2>
      Tic Tac Toe
    </h2>
    <Board>
      <div
        className="board"
        style={
          Object {
            "margin": "10px auto",
            "width": "400px",
          }
        }
      >
        <div
          className="board-wrapper"
        >
          <div
            className="col box"
            key="0"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="1"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="2"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="3"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="4"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="5"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="6"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="7"
            onClick={[Function]}
            style={Object {}}
          />
          <div
            className="col box"
            key="8"
            onClick={[Function]}
            style={Object {}}
          />
        </div>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          Reset
        </button>
        <h3
          className="feedback"
        >
          Enjoy the game! Any feedback? Kindly drop a feedback mail at ezikeonyinyefavour@gmail.com, Thank you!
        </h3>
      </div>
    </Board>
  </div>
</App>
`;
